### PR TITLE
Update clang CI, treat warnings as errors

### PR DESCRIPTION
**Pull request by ghost**
* manual patch

 

**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the documentation and the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**
see https://github.com/rizinorg/rizin/pull/516#issuecomment-775870061

**Test plan**

...

**Closing issues**

...

### CODE REVIEW

File: .github/workflows/clang-analysis.yml [hunk 1]
```diff
@@ -41,6 +41,9 @@ jobs:
     - name: Install npm
       run: sudo apt --assume-yes install npm
 
+    - name: Install npm
```
Comment (Maijin) [inline .github/workflows/clang-analysis.yml]: typo, should be Install jq

File: binrz/rizin/rizin.c [hunk 2]
```diff
@@ -41,6 +41,7 @@ static int rz_main_rzpipe(int argc, const char **argv) {
 		eprintf("Error: RZ_PIPE_(IN|OUT) environment not set\n");
 		eprintf("Usage: rizin -c '!*rzp x' # run commands via rzpipe\n");
 		rc = 1;
+		rc = 1;
```
Comment (Maijin) [inline binrz/rizin/rizin.c]: isn't that dupe?

**Comment by ghost** [comment]
2021-02-10-160008_823x40_scrot

The fix is working

File: binrz/rizin/rizin.c (hunk 2 above)
Comment (ghost) [reply binrz/rizin/rizin.c]: Yes, I want to generate a new warning to check if it is correctly process as an error

**Comment by ghost** [comment]
I am waiting https://github.com/rizinorg/rizin/pull/516

**Comment by ghost** [comment]
Any idea how to refactor the file? This is always the same steps.

**Comment by ret2libc** [comment]
> Any idea how to refactor the file? This is always the same steps.

See ci.yml and how `matrix` is used. You can have two items in the matrix and set some properties for each item. Then you can execute/skip some step in the job based on some properties of the matrix item.

**Review by ret2libc — APPROVED** [review]

**Pull request merged**